### ISSUE TITLE
[Bug] Crash when using xray without sodium on 1.21

### ISSUE THREAD

**Issue by mateooftherice** [MeteorDevelopment/meteor-client#4686]
### Describe the bug

everytime i log onto a server the game crashes with error code -1

```
---- Minecraft Crash Report ----
// Who set us up the TNT?

Time: 2024-06-25 14:08:51
Description: Batching sections

java.lang.IndexOutOfBoundsException
	at java.base/java.nio.Buffer$1.apply(Buffer.java:757)
	at java.base/java.nio.Buffer$1.apply(Buffer.java:754)
	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:213)
	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:210)
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:98)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:106)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:302)
	at java.base/java.nio.Buffer.checkIndex(Buffer.java:768)
	at java.base/java.nio.DirectFloatBufferU.get(DirectFloatBufferU.java:342)
	at net.minecraft.class_9801.method_60820(class_9801.java:39)
	at net.minecraft.class_9801.method_60819(class_9801.java:69)
	at net.minecraft.class_9810.method_60904(class_9810.java:90)
	at net.minecraft.class_846$class_851$class_4578.method_22783(class_846.java:496)
	at net.minecraft.class_846.method_22757(class_846.java:104)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at java.base/java.nio.Buffer$1.apply(Buffer.java:757)
	at java.base/java.nio.Buffer$1.apply(Buffer.java:754)
	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:213)
	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:210)
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:98)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:106)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:302)
	at java.base/java.nio.Buffer.checkIndex(Buffer.java:768)
	at java.base/java.nio.DirectFloatBufferU.get(DirectFloatBufferU.java:342)
	at net.minecraft.class_9801.method_60820(class_9801.java:39)
	at net.minecraft.class_9801.method_60819(class_9801.java:69)
	at net.minecraft.class_9810.method_60904(class_9810.java:90)
	at net.minecraft.class_846$class_851$class_4578.method_22783(class_846.java:496)

-- Uptime --
Details:
	JVM uptime: 66.838s
	Wall uptime: 51.745s
	High-res time: 50.380s
	Client ticks: 808 ticks / 40.400s
Stacktrace:
	at net.minecraft.class_310.method_1587(class_310.java:2501)
	at net.minecraft.class_310.method_43589(class_310.java:931)
	at net.minecraft.class_310.method_54579(class_310.java:944)
	at net.minecraft.class_310.method_1514(class_310.java:873)
	at net.minecraft.client.main.Main.main(Main.java:256)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- Affected level --
Details:
	All players: 1 total; [class_746['Instagram_com'/99981, l='ClientLevel', x=-970.43, y=112.00, z=-36.99]]
	Chunk stats: 729, 453
	Level dimension: minecraft:overworld
	Level spawn location: World: (-224,74,-160), Section: (at 0,10,0 in -14,4,-10; chunk contains blocks -224,-64,-160 to -209,319,-145), Region: (-1,-1; contains chunks -32,-32 to -1,-1, blocks -512,-64,-512 to -1,319,-1)
	Level time: 6403936 game time, 7320562 day time
	Server brand: Spigot
	Server type: Non-integrated multiplayer server
	Tracked entity count: 43

-- Last reload --
Details:
	Reload number: 1
	Reload reason: initial
	Finished: Yes
	Packs: vanilla, fabric, fabric-resource-loader-v0, fabricloader, meteor-client

-- System Details --
Details:
	Minecraft Version: 1.21
	Minecraft Version ID: 1.21
	Operating System: Windows 11 (amd64) version 10.0
	Java Version: 21.0.3, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 128519928 bytes (122 MiB) / 838860800 bytes (800 MiB) up to 2147483648 bytes (2048 MiB)
	CPUs: 20
	Processor Vendor: GenuineIntel
	Processor Name: 12th Gen Intel(R) Core(TM) i7-12700H
	Identifier: Intel64 Family 6 Model 154 Stepping 3
	Microarchitecture: Alder Lake
	Frequency (GHz): 2.69
	Number of physical packages: 1
	Number of physical CPUs: 14
	Number of logical CPUs: 20
	Graphics card #0 name: NVIDIA GeForce RTX 3060 Laptop GPU
	Graphics card #0 vendor: NVIDIA
	Graphics card #0 VRAM (MiB): 6144.00
	Graphics card #0 deviceId: VideoController1
	Graphics card #0 versionInfo: 31.0.15.5212
	Graphics card #1 name: Intel(R) Iris(R) Xe Graphics
	Graphics card #1 vendor: Intel Corporation
	Graphics card #1 VRAM (MiB): 1024.00
	Graphics card #1 deviceId: VideoController2
	Graphics card #1 versionInfo: 31.0.101.3962
	Memory slot #0 capacity (MiB): 8192.00
	Memory slot #0 clockSpeed (GHz): 4.80
	Memory slot #0 type: Unknown
	Memory slot #1 capacity (MiB): 8192.00
	Memory slot #1 clockSpeed (GHz): 4.80
	Memory slot #1 type: Unknown
	Virtual memory max (MiB): 23499.12
	Virtual memory used (MiB): 17973.22
	Swap memory total (MiB): 7424.00
	Swap memory used (MiB): 150.32
	Space in storage for jna.tmpdir (MiB): available: 27292.68, total: 462472.31
	Space in storage for org.lwjgl.system.SharedLibraryExtractPath (MiB): available: 27292.68, total: 462472.31
	Space in storage for io.netty.native.workdir (MiB): available: 27292.68, total: 462472.31
	Space in storage for java.io.tmpdir (MiB): available: 27292.68, total: 462472.31
	Space in storage for workdir (MiB): available: 27292.68, total: 462472.31
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Launched Version: fabric-loader-0.15.11-1.21
	Launcher name: minecraft-launcher
	Backend library: LWJGL version 3.3.3-snapshot
	Backend API: NVIDIA GeForce RTX 3060 Laptop GPU/PCIe/SSE2 GL version 3.2.0 NVIDIA 552.12, NVIDIA Corporation
	Window size: 1920x1200
	GFLW Platform: win32
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: 
	Is Modded: Definitely; Client brand changed to 'fabric'
	Universe: 400921fb54442d18
	Type: Client (map_client.txt)
	Graphics mode: fancy
	Render Distance: 10/12 chunks
	Resource Packs: vanilla, fabric, fabric-resource-loader-v0, fabricloader, meteor-client
	Current Language: en_us
	Locale: en_US
	System encoding: Cp1252
	File encoding: UTF-8
	CPU: 20x 12th Gen Intel(R) Core(TM) i7-12700H

-- Meteor Client --

Version: 0.5.8
Dev Build: 2112

[[ Active Modules ]]

[Combat]:
hitboxes

[Player]:
anti-hunger
reach
speed-mine

[Movement]:
air-jump
entity-control
jesus
no-fall

[Render]:
fullbright
storage-esp
tunnel-esp
unfocused-cpu
waypoints
xray

[Misc]:
better-chat
```
### Steps to reproduce

logged onto a smp

### Meteor Version

meteor 0.5.8

### Minecraft Version

1.21

### Operating System

Windows

### Before submitting a bug report

- [X] This bug wasn't already reported (I have searched bug reports on GitHub).

- [X] This is a valid bug (I am able to reproduce this on the latest dev build).

**Comment by Big-Iron-Cheems** [comment]
Disable x-ray in the main menu or temporarily install Sodium, known issue.

**Comment by Big-Iron-Cheems** [comment]
Fixed in [`cff4cf5`](https://github.com/MeteorDevelopment/meteor-client/commit/cff4cf578c71691fc17bf4e50619975f0200f1fd), update and it should work 👍
